File: src/main/java/com/metlife/santa/core/Dancer.java
package com.metlife.santa.core;

import net.sf.cb2java.copybook.Copybook;
import net.sf.cb2java.copybook.CopybookParser;
import net.sf.cb2java.data.Record;
import org.apache.spark.api.java.function.Function;
import java.io.IOException;
import java.io.Serializable;
import java.util.Map;


public class Dancer extends ReindeerBase implements Serializable {

    public ReindeerBase init(String file) throws IOException {

        initReindeer(file);
        return this;
    }

    @Override
    public void process() {
        try {
            String file=this.prop.getProperty("copybook_url");

            this.rddOutput=rddInput.map(new Function<String,Map>() {
                @Override
                public Map call(String v1) throws Exception {

                    Copybook copybook = CopybookParser.parse("A", this.getClass().getClassLoader().getResourceAsStream(file));
                    Record record=copybook.parseData(v1.getBytes());

                    return record.toMap();
                }
            });

        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void chain(ReindeerBase previous){
        this.rddInput=previous.rddOutput;
    }
}

File: src/main/scala/com/metlife/santa/core/Dancer.scala
package com.metlife.santa.core

import java.io.{File, FileInputStream}
import net.sf.cb2java.copybook.{Copybook, CopybookParser}
import org.apache.spark.rdd.RDD

class Dancer extends ReindeerBase{

  override def init(file: String):ReindeerBase = {
    initReindeer(file)
    this
  }

  override def process() = {

    println("")
    println("=============== DANCER ( COPYBOOK 2 MAP ) ===================")
    println("")

    val _tempRDD=inputRDD.asInstanceOf[RDD[String]]

    val file: String = this.prop.getProperty("copybook_url")


    spark.sparkContext.textFile("/Users/abhisekjana/spark/reindeer/src/main/resources/a.input.txt").collect().foreach(println)



    val copybook=spark.sparkContext.broadcast[Copybook](CopybookParser.parse("A", new FileInputStream(new File(file))))



    /*inputRDD.foreachPartition{records =>
      records.map(record =>
        copybook.value.parseData(record.getBytes).toMap()
      )
    }*/

    outputRDD=_tempRDD.map(line=>{
      val record=copybook.value.parseData(line.getBytes())
      record.toMap()

    })

    //outputRDD.collect().foreach(println)

  }
}
